kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/lemmas.k

```diff
@@ -99,6 +90,10 @@ module LEMMAS-WITHOUT-SLOT-UPDATES [symbolic]
     rule [b2w-bxor-one]:            1  xorInt bool2Word(Y) => bool2Word(notBool Y)                                             [simplification, comm]
     rule [b2w-bxor]:      bool2Word(X) xorInt bool2Word(Y) => bool2Word ( (X andBool notBool Y) orBool (notBool X andBool Y) ) [simplification]
 
+    // Relationship with `#rangeBool`
+    rule [b2w-rangeBool-eq-not-zero-l]: bool2Word (notBool (X ==Int 0)) => X requires #rangeBool(X) [simplification, comm]
+    rule [b2w-rangeBool-eq-not-zero-r]: bool2Word (notBool (0 ==Int X)) => X requires #rangeBool(X) [simplification, comm]
+
```

Comment:
Do we also want the ones for `X ==Int 1` and `1 ==Int X`?

Reply:
I don't think so, at least not for now, because these ones (the one with `X ==Int 0` specifically, and then I added the commutative version) come from the compiler as a check that something is a bool.